lib/params/MapperFunction.cpp

```diff
@@ -300,25 +310,43 @@ void MapperFunction::hsvValue(float value, float *h, float *s, float *v) const
   }
 }
 
+void MapperFunction::checkForOpacityChanges() {
+	for (int i=0; i<getNumOpacityMaps(); i++) {
+		OpacityMap* om = GetOpacityMap(i);
+		if (om->GetDirtyBit())
+			_dirtyBit = true;
+		om->ResetDirtyBit();
+	}
+}
+
 //----------------------------------------------------------------------------
 // Populate at a RGBA lookup table 
 //----------------------------------------------------------------------------
-void MapperFunction::makeLut(float* clut) const
+void MapperFunction::makeLut(float* clut) 
 {
-  float step = (getMaxMapValue() - getMinMapValue())/float(_numEntries-1);
-
-  for (int i = 0; i< _numEntries; i++)
-  {
-    float v = getMinMapValue() + i*step;
-    m_colorMap->color(v).toRGB(&clut[4*i]);
-    clut[4*i+3] = getOpacityValueData(v);
+  checkForOpacityChanges();
+
+  if (_dirtyBit) {
+  	float step = (getMaxMapValue() - getMinMapValue())/float(_numEntries-1);
+  	for (int i = 0; i< _numEntries; i++)
+  	{
+  	  float v = getMinMapValue() + i*step;
+  	  m_colorMap->color(v).toRGB(&clut[4*i]);
+  	  clut[4*i+3] = getOpacityValueData(v);
+  	}
+	memcpy(_clut, clut, sizeof(float)*_numEntries*4);
+  }
+  else {
+	memcpy(clut, _clut, sizeof(float)*_numEntries*4);
   }
+
+  _dirtyBit = false;
```

Comment:
Should this `_dirtyBit = false;` assignment be in the bracket of line 329?

Reply:
No.  Once we regenerate the lookup table we reset the dirty bit.